src/LinuxTracing/ThreadStateManager.h

```diff
@@ -60,10 +62,26 @@ class ThreadStateManager {
 
  private:
   struct OpenState {
-    OpenState(orbit_grpc_protos::ThreadStateSlice::ThreadState state, uint64_t begin_timestamp_ns)
-        : state{state}, begin_timestamp_ns{begin_timestamp_ns} {}
+    OpenState(orbit_grpc_protos::ThreadStateSlice::ThreadState state, uint64_t begin_timestamp_ns,
+              pid_t wakeup_tid = orbit_base::kInvalidThreadId,
+              pid_t wakeup_pid = orbit_base::kInvalidProcessId,
+              orbit_grpc_protos::ThreadStateSlice_WakeupReason wakeup_reason = orbit_grpc_protos::
+                  ThreadStateSlice_WakeupReason::ThreadStateSlice_WakeupReason_kNotApplicable)
+        : state{state},
+          begin_timestamp_ns{begin_timestamp_ns},
+          wakeup_tid{wakeup_tid},
+          wakeup_pid{wakeup_pid},
+          wakeup_reason{wakeup_reason} {}
     orbit_grpc_protos::ThreadStateSlice::ThreadState state;
     uint64_t begin_timestamp_ns;
+    // The next two fields are optional. We use them to transfer info between interruptible sleep
+    // state (gray) and runnable state (blue), so we can know which tid and pid caused the thread to
+    // turn into a runnable state.
+    pid_t wakeup_tid = orbit_base::kInvalidThreadId;
+    pid_t wakeup_pid = orbit_base::kInvalidProcessId;
+    // the following field explains the relation between this thread and the thread that woke it up.
+    orbit_grpc_protos::ThreadStateSlice_WakeupReason wakeup_reason = orbit_grpc_protos::
```

Comment:
I think the type can also be access via `orbit_grpc_protos::ThreadStateSlice::WakeupReason`. That's what we do for `orbit_grpc_protos::ThreadStateSlice::ThreadState` which is also a protobuf enum. I would prefer to be consistent here and replace `ThreadStateSlice_WakeupReason` by `ThreadStateSlice::WakeupReason` everywhere.

Reply:
(You should be able to access `kNotApplicable` via `orbit_grpc_protos::ThreadStateSlice::WakeupReason::kNotApplicable` which is also shorter and more readable than `orbit_grpc_protos::ThreadStateSlice_WakeupReason::ThreadStateSlice_WakeupReason_kNotApplicable`)